source/frontend/src/features/documents/hooks/useDocumentProvider.ts

```diff
@@ -90,6 +95,31 @@ export const useDocumentProvider = () => {
       }, []),
     });
 
+  // Provides functionality for uploading a document metadata
+  const { execute: updateDocument, loading: updateDocumentLoading } = useApiRequestWrapper<
+    (
+      documentId: number,
+      updateRequest: Api_DocumentUpdateRequest,
+    ) => Promise<AxiosResponse<Api_DocumentUpdateResponse, any>>
+  >({
+    requestFunction: useCallback(
+      async (documentId: number, updateRequest: Api_DocumentUpdateRequest) =>
+        await updateDocumentMetadataApiCall(documentId, updateRequest),
+      [updateDocumentMetadataApiCall],
+    ),
+    requestName: 'updateDocumentMetadataApiCall',
+    onSuccess: useCallback(() => {
+      toast.success('Updated document metadata');
+    }, []),
+    onError: useCallback((axiosError: AxiosError<IApiError>) => {
+      if (axiosError?.response?.status === 400) {
+        toast.error(axiosError?.response.data.error);
+      } else {
+        toast.error('Update document error. Check responses and try again.');
```

Comment:
Update document error or Update document metadata error?

Reply:
Document. Although the call has metadata on its name, it is possible that the status is the only part being updated